test_mypy fails

Something changed around mypy and our tests are failing: https://github.com/python-attrs/attrs/runs/4030751250?check_suite_focus=true#step:5:57

cc @euresti 

(sorry for terseness, I'm on the jump rn)

<details>
```
 /home/runner/work/attrs/attrs/tests/test_mypy.yml:784: 
59
  E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output: 
60
  E   Actual:
61
  E     main:15: error: Cannot determine __init__ type from converter (diff)
62
  E     main:15: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]" (diff)
63
  E     main:16: error: Cannot determine __init__ type from converter (diff)
64
  E     main:16: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]" (diff)
65
  E     ...
66
  E   Expected:
67
  E     main:15: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]" (diff)
68
  E     main:15: error: Cannot determine __init__ type from converter (diff)
69
  E     main:16: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]" (diff)
70
  E     main:16: error: Cannot determine __init__ type from converter (diff)
71
  E     ...
72
  E   Alignment of first line difference:
73
  E     E: main:15: error: Argument "converter" has incompatible type "Callable[[],...
74
  E     A: main:15: error: Cannot determine __init__ type from converter...
75
  E                        ^
76
  _____________________ testAttrsUsingBadConverterReprocess ______________________
77
  /home/runner/work/attrs/attrs/tests/test_mypy.yml:813: 
78
  E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output: 
79
  E   Actual:
80
  E     main:16: error: Cannot determine __init__ type from converter (diff)
81
  E     main:16: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]" (diff)
82
  E     main:17: error: Cannot determine __init__ type from converter (diff)
83
  E     main:17: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]" (diff)
84
  E     ...
85
  E   Expected:
86
  E     main:16: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]" (diff)
87
  E     main:16: error: Cannot determine __init__ type from converter (diff)
88
  E     main:17: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]" (diff)
89
  E     main:17: error: Cannot determine __init__ type from converter (diff)
90
  E     ...
91
  E   Alignment of first line difference:
92
  E     E: main:16: error: Argument "converter" has incompatible type "Callable[[],...
93
  E     A: main:16: error: Cannot determine __init__ type from converter...
94
  E                        ^
```
</details>

--- a/tests/test_mypy.yml
+++ b/tests/test_mypy.yml
@@ -785,10 +785,10 @@
         bad_overloaded: int = attr.ib(converter=bad_overloaded_converter)
     reveal_type(A)
   out: |
-    main:15: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]"
     main:15: error: Cannot determine __init__ type from converter
-    main:16: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]"
+    main:15: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]"
     main:16: error: Cannot determine __init__ type from converter
+    main:16: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]"
     main:17: note: Revealed type is "def (bad: Any, bad_overloaded: Any) -> main.A"
 
 - case: testAttrsUsingBadConverterReprocess
@@ -814,10 +814,10 @@
         bad_overloaded: int = attr.ib(converter=bad_overloaded_converter)
     reveal_type(A)
   out: |
-    main:16: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]"
     main:16: error: Cannot determine __init__ type from converter
-    main:17: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]"
+    main:16: error: Argument "converter" has incompatible type "Callable[[], str]"; expected "Callable[[Any], Any]"
     main:17: error: Cannot determine __init__ type from converter
+    main:17: error: Argument "converter" has incompatible type overloaded function; expected "Callable[[Any], Any]"
     main:18: note: Revealed type is "def (bad: Any, bad_overloaded: Any) -> main.A"
 
 - case: testAttrsUsingUnsupportedConverter

